### PR TITLE
chore: drop support for license files

### PR DESCRIPTION
**Pull request by thalesmg**
#8952 but against master

Related issue: EMQX-7247

EMQX ee5.0 is to drop support for both the old license format (x509) and for files containing the new format.

### CODE REVIEW

File: lib-ee/emqx_license/etc/emqx_license.conf [hunk 1]
```diff
@@ -1,5 +1,4 @@
 license {
-    type = key
     # The default license has 1000 connections limit, it is issued on 20220419 and valid for 5 years (1825 days)
     key = "MjIwMTExCjAKMTAKRXZhbHVhdGlvbgpjb250YWN0QGVtcXguaW8KZGVmYXVsdAoyMDIyMDQxOQoxODI1CjEwMDAK.MEQCICbgRVijCQov2hrvZXR1mk9Oa+tyV1F5oJ6iOZeSHjnQAiB9dUiVeaZekDOjztk+NCWjhk4PG8tWfw2uFZWruSzD6g=="
```
Comment (zmstone) [inline lib-ee/emqx_license/etc/emqx_license.conf:3]: delete this default config.

File: lib-ee/emqx_license/etc/emqx_license.conf [hunk 2]
```diff
@@ -1,5 +1,4 @@
 license {
-    type = key
     # The default license has 1000 connections limit, it is issued on 20220419 and valid for 5 years (1825 days)
```
Comment (zmstone) [inline lib-ee/emqx_license/etc/emqx_license.conf:2]: move this to i18n

also: delete the 'type' description in i18n

File: lib-ee/emqx_license/etc/emqx_license.conf [hunk 3]
```diff
@@ -1,5 +1,4 @@
...
     key = "MjIwMTExCjAKMTAKRXZhbHVhdGlvbgpjb250YWN0QGVtcXguaW8KZGVmYXVsdAoyMDIyMDQxOQoxODI1CjEwMDAK.MEQCICbgRVijCQov2hrvZXR1mk9Oa+tyV1F5oJ6iOZeSHjnQAiB9dUiVeaZekDOjztk+NCWjhk4PG8tWfw2uFZWruSzD6g=="
     connection_low_watermark = 75%,
```
Comment (zmstone) [inline lib-ee/emqx_license/etc/emqx_license.conf:4]: if we have default values for these two, we may as well just delete this entire file ?

File: lib-ee/emqx_license/etc/emqx_license.conf (hunk 2 above)
Comment (thalesmg) [reply lib-ee/emqx_license/etc/emqx_license.conf:2]: I think I deleted the type description in i18n? https://github.com/emqx/emqx/pull/8961/files#diff-a42874a345b13646d792ddcffaae95be3972e37b4d850358d9a6f85c827d6b61L23

Did it slip past somewhere else?  I couldn't find mentions of type there :see_no_evil:

Comment (zmstone) [reply lib-ee/emqx_license/etc/emqx_license.conf]: Sorry, I missed it.

Please move this comment to the `key_field` description in i18n.

File: lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf [hunk 4]
```diff
@@ -2,15 +2,13 @@ emqx_license_schema {
   license_root {
     desc {
       en: "Defines the EMQX Enterprise license. \n\n"
-          "A license is either a `key` or a `file`.\n"
-          "When `key` and `file` are both configured, `key` is used.\n"
+          "\n"
+          "The default license has 1000 connections limit, it is "
```
Comment (zmstone) [inline lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf:6]: maybe better to keep it close to the default value.

File: lib-ee/emqx_license/etc/emqx_license.conf (hunk 2 above)
Comment (thalesmg) [reply lib-ee/emqx_license/etc/emqx_license.conf]: :heavy_check_mark:

File: lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf [hunk 5]
```diff
@@ -2,15 +2,13 @@ emqx_license_schema {
...
+          "issued on 2022-04-19 and valid for 5 years (1825 days).\n"
           "\n"
           "EMQX comes with a default trial license.  For production use, please \n"
           "visit https://www.emqx.com/apply-licenses/emqx to apply."
       zh: "EMQX企业许可证。\n"
-          "许可证是一个 `key` 或一个 `file`。\n"
-          "当 `key` 和 `file` 同时被配置时，优先使用 `key`。\n"
-          "\n"
           "EMQX 自带一个默认的试用许可证，若需要在生产环境部署，\n"
```
Comment (zmstone) [inline lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf:12]:
```suggestion
          "EMQX 自带一个默认的试用许可证，"
          "默认试用许可允许最多接入 1000 个连接，签发时间是 2022年4月19日，有效期是 5 年（1825 天）。"
          "若需要在生产环境部署，\n"
```

Comment (thalesmg) [reply lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf]: Oops, had forgotten to translate that :see_no_evil: 
Thanks!  :beers:

File: lib-ee/emqx_license/src/emqx_license_schema.erl [hunk 6]
```diff
@@ -34,55 +32,18 @@ roots() ->
...
 fields(key_license) ->
     [
-        {type, #{
-            type => key,
-            required => true,
-            desc => ?DESC(license_type_field)
-        }},
         {key, #{
             type => string(),
+            default =>
```
Comment (zmstone) [inline lib-ee/emqx_license/src/emqx_license_schema.erl:37]: nit: move this to a function in emqx_license module.
```
%% @doc The default license key.
%% This default license has 1000 connections limit.
%% It is issued on 2022-04-19 and valid for 5 years (1825 days)
%% NOTE: when updating a new key, schema doc in lib-ee/emqx_license/i18n/emqx_license_schema_i18n.conf should be updated accordingly
default_license() ->
    "MjIwMTExCjAKMTAKRXZhbHVhdGlvbgpjb250YWN0QGVtcXguaW8KZ"
    "GVmYXVsdAoyMDIyMDQxOQoxODI1CjEwMDAK.MEQCICbgRVijCQov2"
    "hrvZXR1mk9Oa+tyV1F5oJ6iOZeSHjnQAiB9dUiVeaZekDOjztk+NC"
    "Wjhk4PG8tWfw2uFZWruSzD6g==",
```

File: lib-ee/emqx_license/test/emqx_license_test_lib.erl [hunk 7]
```diff
@@ -82,9 +77,6 @@ make_license(Values) ->
     iolist_to_binary([EncodedText, ".", EncodedSignature]).
 
 default_license() ->
-    %% keep it the same as in etc/emqx_license.conf
-    License =
-        "MjIwMTExCjAKMTAKRXZhbHVhdGlvbgpjb250YWN0QGVtcXguaW8KZGVmYXVsdAoyMDIyMDQxOQoxODI1CjEwMDAK."
-        "MEQCICbgRVijCQov2hrvZXR1mk9Oa+tyV1F5oJ6iOZeSHjnQAiB9dUiVeaZekDOjztk+NCWjhk4PG8tWfw2uFZWruSzD6g==",
-    ok = file:write_file(?DEFAULT_LICENSE_FILE, License),
-    ?DEFAULT_LICENSE_FILE.
+    %% keep it the same as in emqx_license_schema.erl
```
Comment (zmstone) [inline lib-ee/emqx_license/test/emqx_license_test_lib.erl:80]: if we move it to `emqx_license.erl`, then there is no need to duplicate here.

**Comment by savonarola** [comment]
+71 −1,078 👍 The things are much simpler now

**Review by savonarola — APPROVED** [review]

File: lib-ee/emqx_license/src/proto/emqx_license_proto_v2.erl [hunk 8]
```diff
@@ -11,20 +11,15 @@
...
-    save_and_backup_license/2
+    remote_connection_counts/1
 ]).
 
 -define(TIMEOUT, 500).
 -define(BACKUP_TIMEOUT, 15_000).
 
 introduced_in() ->
-    "5.0.5".
+    "e5.0.0".
```
Comment (savonarola) [inline lib-ee/emqx_license/src/proto/emqx_license_proto_v2.erl:21]: Shouldn't be the proto index file be updated?

Comment (thalesmg) [reply lib-ee/emqx_license/src/proto/emqx_license_proto_v2.erl:21]: Maybe surprisingly, no alarms went of in the static checks. :thinking: 

Also, I think that [`introduced_in`](https://github.com/emqx/emqx/blob/982abd96bf199e82bee25f351797539f71031472/apps/emqx/src/bpapi/emqx_bpapi_trans.erl#L67) is currently purely informational, not being recorded anywhere.

**Pull request merged**